### PR TITLE
[aptos-rosetta] Chain Construction APIs together via testing and add better errors

### PR DESCRIPTION
**Pull request by gregnazario**
### Description
This gets Construct working for both CreateAccount and Transfer operations to the Rosetta spec.

Additionally, I cleaned up the errors for Rosetta in general, fixed some bugs, added parsing for operations.

There are a few TODOs leftover for getting the genesis hash and adding peers.  Otherwise, the entire system does work with my test framework.  Additionally, I'll need to build the Rosetta DSL to finish the testing.

Built on top of https://github.com/aptos-labs/aptos-core/pull/1776

### Test Plan
Create flow:

```
$ aptos account fund --profile local --account local --num-coins 100000000
{
  "Result": "Added 100000000 coins to account CB48471868293A5FEB8FAC3871775B62A671B7C405B43E7755F368901DF9EC8C"
}

$ curl localhost:8080/accounts/0xcb48471868293a5feb8fac3871775b62a671b7c405b43e7755f368901df9ec8c/resources 2> /dev/null | jq | grep -A 1 coin
      "coin": {
        "value": "100000000"

$ cargo run -p aptos-rosetta-cli -- construction create-account --new-account 0x1337 --profile local --sender local                                           
    Finished dev [unoptimized + debuginfo] target(s) in 0.30s
     Running `target/debug/aptos-rosetta-cli construction create-account --new-account 0x1337 --profile local --sender local`
{
  "hash": "0xbd5433448fa3c761c2be3f6527713caa293572f42036789d8a177d8ce04a7c31"
}

$ curl localhost:8080/accounts/0xcb48471868293a5feb8fac3871775b62a671b7c405b43e7755f368901df9ec8c/resources 2> /dev/null | jq | grep -A 1 coin
      "coin": {
        "value": "99999861"
$ curl localhost:8080/accounts/0x1337
{"sequence_number":"0","authentication_key":"0x0000000000000000000000000000000000000000000000000000000000001337"}%             
```

Transfer flow:
```
$ ./target/debug/aptos-rosetta-cli construction transfer --profile local --sender local --receiver 0x1337 --amount 10000
{
  "hash": "0xb534a43fe98c0d43883bf8ac90f06c9749434a241e42930a97f82918b9c68b48"
}

$ curl localhost:8080/accounts/0x1337
{"sequence_number":"0","authentication_key":"0x0000000000000000000000000000000000000000000000000000000000001337"}% 

$ curl localhost:8080/accounts/0xcb48471868293a5feb8fac3871775b62a671b7c405b43e7755f368901df9ec8c/resources 2> /dev/null | jq | grep -A 1 coin
      "coin": {
        "value": "99989788"
        
$ curl localhost:8080/accounts/0x1337/resources 2> /dev/null | jq | grep -A 1 coin
      "coin": {
        "value": "10000"
```

---
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/1781)

### CODE REVIEW

File: crates/aptos-rosetta-cli/src/construction.rs [hunk 1]
```diff
@@ -0,0 +1,352 @@
+// Copyright (c) Aptos
```
Comment (davidiw) [inline crates/aptos-rosetta-cli/src/construction.rs:1]: maybe some documentation / help guidance?

File: crates/aptos-rosetta/src/client.rs [hunk 2]
```diff
@@ -30,80 +40,102 @@ impl RosettaClient {
...
+        self.make_call("construction/metadata", request).await
+    }
+
+    pub async fn parse(
+        &self,
+        request: &ConstructionParseRequest,
+    ) -> anyhow::Result<ConstructionParseResponse> {
+        self.make_call("construction/parse", request).await
+    }
+    pub async fn payloads(
```
Comment (davidiw) [inline crates/aptos-rosetta/src/client.rs:84]: nit new line

File: crates/aptos-rosetta/src/client.rs [hunk 3]
```diff
@@ -30,80 +40,102 @@ impl RosettaClient {
...
+        &self,
+        request: &ConstructionPayloadsRequest,
+    ) -> anyhow::Result<ConstructionPayloadsResponse> {
+        self.make_call("construction/payloads", request).await
+    }
+    pub async fn preprocess(
```
Comment (davidiw) [inline crates/aptos-rosetta/src/client.rs:90]: cough

File: crates/aptos-rosetta/src/types/move_types.rs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+pub fn create_account_identifier() -> Identifier {
+    ident_str!("create_account").into()
+}
+
+pub fn test_coin_identifier() -> Identifier {
+    ident_str!("TestCoin").into()
+}
+pub fn sequence_number_identifier() -> Identifier {
```
Comment (davidiw) [inline crates/aptos-rosetta/src/types/move_types.rs:28]: mind the gap

**Review by davidiw — APPROVED** [review]
we should probably invest in more documentation, I worry you'll be the owner of this for the rest of your life.

File: crates/aptos-rosetta-cli/src/construction.rs (hunk 1 above)
Comment (gregnazario) [reply crates/aptos-rosetta-cli/src/construction.rs:1]: Done.

**Pull request merged**